Adds support to choose between JLINK and BMP as programmers

EarthLord opened the pull request:
The targets of flash, flash_softdevice and erase work based
on what the variable DEBUGGER is set as: JLINK or BMP. 
Needs `arm-none-eabi-gdb` in the path for BMP to work.

EarthLord committed:
Adds support to choose between JLINK and BMP as programmers

The targets of flash, flash_softdevice and erase work based
on what the variable DEBUGGER is set as: JLINK or BMP.

electronut commented:
You seem to have hardcoded the USB port:

>  /dev/ttyBmpGdb

That will not work on Windows and Linus. Instead, it should be a `make` variable.

Eg.:

> $make BMP_PORT=COM5 flash

Also I think we need a check on DEBUGGER - if it's already from cmd line, it should take precedence. eg.:

> $make DEBUGGER=BMP BMP_PORT=COM5 flash

EarthLord commented:
What is this 'Windows' that you speak of? :P 

OK, I'll see what I can do for making it compatible with both the OS. Also there is an issue with the path of nrfjprog. I'll fix that too.

electronut commented:
`Windows` - holds inmates. :-)

EarthLord committed:
Merge branch 'master' of github.com:electronut/ElectronutLabs-bluey

EarthLord committed:
Adds Makefile variable for BMP port and nrfjprog path

This commit removes the hardcoding of the BMP port to /dev/ttyBmpGdb
so that people using other OS apart from Linux can set it appropriately.

EarthLord commented:
Ok, I've made the changes. Untested in Windows and Mac though.
